### PR TITLE
New Rollback Command, comprehensive Test Suite & Documentation Enhancement

### PR DESCRIPTION
**Pull request by Tehem**
# 🧪✨ New Rollback Command, Comprehensive Test Suite & Documentation Enhancement + Rollback Feature

## 📝 Description
This PR introduces a comprehensive test suite, enhances documentation, and adds a new rollback feature to improve code quality, maintainability, and safety. The changes include:

### 🎯 Key Changes
- Added test coverage for core components:
  - `libs/solver/*` - Testing solver logic and algorithms
  - `libs/types/*` - Testing data structures and types
  - `libs/timeutil/*` - Testing time-related utilities
- Added test utilities in `libs/testutils/` for easier test setup
- Enhanced README with detailed testing documentation
- Implemented new `rollback` command for safe event management

### 🔄 New Rollback Command
- Added ability to rollback calendar events by batch ID
- Safe deletion of events with detailed logging
- Interactive mode for user-friendly operation
- Batch file management for tracking created events
- Comprehensive error handling and reporting

### 🧪 Test Coverage
- Unit tests for all major components
- Test utilities for configuration mocking
- Comprehensive test cases for edge cases
- Improved test readability and maintainability

### 📚 Documentation
- Added detailed testing section in README
- Documented various test commands and their usage
- Added examples for common testing scenarios
- Added rollback command usage examples

### 🚀 Impact
- Improved code reliability through comprehensive testing
- Better developer experience with clear testing documentation
- Easier maintenance with well-documented test cases
- Enhanced confidence in code changes through test coverage
- Added safety net with rollback capability for calendar events

### 🔍 Testing Instructions
```bash
# Run all tests
go test ./...

# Run tests with coverage
go test -cover ./...

# Generate coverage report
go test -coverprofile=coverage.out ./...
go tool cover -html=coverage.out
```

### 🔄 Rollback Usage Examples
```bash
# Rollback with batch ID
matchmaker rollback 123e4567-e89b-12d3-a456-426614174000

# Interactive rollback
matchmaker rollback
Enter batch ID: 123e4567-e89b-12d3-a456-426614174000
```

### 📋 Checklist
- [x] Added unit tests for all major components
- [x] Created test utilities for common test scenarios
- [x] Enhanced README with testing documentation
- [x] Implemented rollback command functionality
- [x] Added rollback command documentation
- [x] Ensured all tests are passing
- [x] Followed Go testing best practices

---

**Note**: This PR significantly improves the project's test coverage, documentation, and safety features, making it more maintainable, reliable, and safer for future development.

### CODE REVIEW

File: libs/types/ranges.go [hunk 1]
```diff
@@ -28,29 +62,43 @@ func (r *Range) Minutes() float64 {
...
+			totalDuration := workRange.Duration()
+			numRanges := int(totalDuration / duration)
+			if numRanges == 0 {
+				continue
+			}
+
+			// Generate ranges
 			start := workRange.Start
-			for !workRange.End.Before(start.Add(duration)) {
+			for range numRanges {
```
[the remark this answers is not in the document]
Comment (Tehem) [reply libs/types/ranges.go:81]: we can do that with go >= 1.22
